### PR TITLE
[BEAM-473] DatastoreIO SplitQueryFn integration test

### PR DESCRIPTION
**Pull request by vikkyrk**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by vikkyrk** [comment]
R: @dhalperi caught a bug :). Let me know what you think

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/datastore/SplitQueryFnIT.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+   * Tests {@link SplitQueryFn} to generate expected number of splits for a large dataset.
+   */
+  @Test
+  public void testSplitQueryFnWithLargeDataset() throws Exception {
+    String projectId = "deft-testing-integration2";
+    String kind = "input_sort_1G";
+    String namespace = null;
+    // Num splits is computed based on the entity_bytes size of the input_sort_1G kind reported by
+    // Datastore stats.
+    int exceptedNumSplits = 19;
```
Comment (dhalperi) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/datastore/SplitQueryFnIT.java]: expected

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/datastore/SplitQueryFnIT.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+/**
+ * Integration tests for {@link DatastoreV1.Read.SplitQueryFn}.
+ *
+ * Note: It is hard to mock the exact behavior of Datastore, especially for the statistics queries.
+ * Also the fact that DatastoreIO falls back gracefully when querying statistics fails, makes it
+ * hard to catch these issues in production. This test here ensures we interact with the Datastore
+ * directly, query the actual stats and verify that the SplitQueryFn generates the expected number
+ * of query splits.
+ */
```
Comment (dhalperi) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/datastore/SplitQueryFnIT.java]: Add comments here saying something like:

```
<p>These tests are brittle as they rely on golden data in Datastore. If the data gets lost or changes then they will begin failing and this test should be disabled.

<p>At the time of writing, the Datastore has the following statistics and these numbers...
```

**Comment by dhalperi** [comment]
LGTM, merged.

**Pull request closed without merging**